### PR TITLE
Fix CI warnings at their source: OBB angle attribute, DDP broadcast_buffers, LiteRT training mode

### PR DESCRIPTION
**Pull request by glenn-jocher**
Reviewed every job log of the latest main CI run (35210762526) and fixed the warnings that originate in our code. Nothing is suppressed or filtered.

## Fixes

| Warning in CI | Cause | Fix |
|---|---|---|
| `UserWarning: The tensor attribute self.model.23.angle was assigned during export. Such attributes must be registered as buffers` (ExecuTorch OBB, all Benchmarks jobs) | `OBB._inference` stashed `x["angle"]` on `self` only so `decode_bboxes` could read it back | Pass the angle to `decode_bboxes` as an argument and delete the attribute. It replaces the `xywh` parameter of `Detect.decode_bboxes`, which no caller ever passed |
| `FutureWarning: broadcast_buffers is deprecated. Use forward_sync_buffers instead` (GPU job, torch 2.14, DDP) | `trainer.py` passes the kwarg deprecated in torch 2.13 | Use `forward_sync_buffers=False` on torch>=2.13, `broadcast_buffers=False` below. The only behavior difference is a buffer sync at init, which is a no-op here because every rank builds the same model from the same weights |
| `WARNING:root:Your model is converted in training mode` (LiteRT detect/segment/pose/obb) | `_NormalizeCoords` (and `NMSModel`) are new `nn.Module` wrappers, which default to `training=True` even around an eval model | `litert_torch.convert(model.eval(), ...)`, same idiom as `coreml.py` |

## Validation

- torch 2.13.0: `torch.export.export` of yolo26n-obb emits the angle warning on main and none on this branch.
- Raw outputs of yolo26n-obb, yolo11n-obb and yolo26n are bit-identical to main on torch 2.9.1 and 2.13.0.
- DDP (gloo): torch 2.13.0 warns with `broadcast_buffers` and is silent with `forward_sync_buffers`; torch 2.9.1 keeps `broadcast_buffers`.
- LiteRT 0.9.4: training-mode warning count 1 on main, 0 here; detect and OBB `.tflite` exports succeed and run inference.
- OBB ONNX and TorchScript export plus inference round trip OK; `pytest tests/test_python.py tests/test_exports.py -k "obb or torchscript or onnx"` 15 passed.

## Left alone, with reasons

- `TracerWarning` at `head.py` 211/254/269 and `transformer.py` 574/578: Python branching on shapes during `torch.jit.trace`. The shapes are meant to be constants in those static exports, so there is no behavior to fix.
- `torch.jit.trace`/`torch.jit.load` deprecations and the legacy TorchScript ONNX exporter notice: inherent to the TorchScript format and the deliberate `dynamo=False` ONNX path.
- `LeafSpec` FutureWarning (163x), torchao/coremltools/MNN `SyntaxWarning`, `pkg_resources`, MCT divide warnings, `KernelPreference` enum notice: raised inside third-party packages.
- `'half'/'int8' is deprecated`: emitted on purpose by the deprecation tests.
- `omnimalloc==0.5.0 is yanked`: pinned alongside the Axelera SDK in `axelera.py` and `EXPORT_ENVS`; moving it needs an Axelera compile run to verify, so it is not changed here.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary

Updated OBB decoding, distributed training setup, and LiteRT export to eliminate warnings originating in Ultralytics code without changing model outputs.

### 📊 Key Changes

- Passes OBB angle tensors directly to `decode_bboxes` instead of storing them as the `OBB.angle` module attribute; the same argument is now supported by the TensorFlow decode path.
- Selects `forward_sync_buffers=False` for PyTorch 2.13 and newer, while retaining `broadcast_buffers=False` for older versions.
- Converts LiteRT models with `model.eval()` so export wrappers do not enter training mode by default.
- Adds the `TORCH_2_13` version check used by the DDP configuration.

### 🎯 Purpose & Impact

- Removes the OBB export warning about assigning tensor attributes during export.
- Removes the DDP deprecation warning on PyTorch 2.13 and newer while preserving the existing setting on older PyTorch versions.
- Removes LiteRT training-mode warnings for detect, segment, pose, and OBB exports.
- OBB decoding and exported model outputs remain functionally unchanged; this is otherwise a maintenance-focused change with no user-facing API change.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed all five changed files and traced the OBB inference/export paths, DDP setup, and LiteRT conversion. The warning fixes are otherwise sound, but the new unconditional angle argument introduces a compatibility regression for existing/custom Detect subclasses. Remaining risk is limited to that changed decode method contract.

<details><summary>💬 Posted 1 inline comment</summary>

- 💡 **MEDIUM** `ultralytics/nn/modules/head.py:215` This unconditionally changes the dispatch contract for every `Detect` subclass: an existing custom head overriding `decode_bboxes(self, bboxes, anchors)` now receives a third positional argument during normal inference and raises `TypeError`. The previous `xywh` parameter on the base method did not require callers to pass it. Pass `angle` only when the prediction dictionary contains it, preserving the two-argument call for non-OBB heads; apply the same conditional dispatch in `tensorflow.py`.
</details>

File: ultralytics/nn/modules/head.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def _get_decode_boxes(self, x: dict[str, torch.Tensor]) -> torch.Tensor:
             self.anchors, self.strides = (a.transpose(0, 1) for a in make_anchors(x["feats"], self.stride, 0.5))
             self.shape = shape
 
-        dbox = self.decode_bboxes(self.dfl(x["boxes"]), self.anchors.unsqueeze(0)) * self.strides
+        dbox = self.decode_bboxes(self.dfl(x["boxes"]), self.anchors.unsqueeze(0), x.get("angle")) * self.strides
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/modules/head.py:215]: 💡 **MEDIUM**: This unconditionally changes the dispatch contract for every `Detect` subclass: an existing custom head overriding `decode_bboxes(self, bboxes, anchors)` now receives a third positional argument during normal inference and raises `TypeError`. The previous `xywh` parameter on the base method did not require callers to pass it. Pass `angle` only when the prediction dictionary contains it, preserving the two-argument call for non-OBB heads; apply the same conditional dispatch in `tensorflow.py`.

Comment (glenn-jocher) [reply ultralytics/nn/modules/head.py:215]: Keeping the unconditional argument. `decode_bboxes` is an internal head hook with exactly two implementations (`Detect`, `OBB`) and three call sites (`head.py`, `tensorflow.py`, `imx.py`). A search of the Ultralytics org and our other repos (portal, sdk, inference, hub, lite, replicate, assistant, actions) finds no subclass overriding it and no external caller. The IMX two-argument call only reaches `Detect.decode_bboxes` (IMX supports detect, pose, segment), where `angle` defaults to `None`. Conditional dispatch at each call site would add branching to guard a subclass that does not exist, and the old `self.angle` side channel is what caused the torch.export warning in the first place.

**Comment by UltralyticsAssistant** [comment]
🎉 [PR #26215](https://github.com/ultralytics/ultralytics/pull/26215) has been merged—thank you @glenn-jocher!

> “Great things are done by a series of small things brought together.” — Vincent van Gogh

This focused maintenance work removes OBB, DDP, and LiteRT export warnings while preserving model outputs and existing user-facing behavior.
